dotnet/WPF/VisualLayerIntegration/VisualLayerIntegration/CompositionHost.cs

```diff
@@ -220,42 +185,21 @@ internal struct DispatcherQueueOptions
             public DISPATCHERQUEUE_THREAD_APARTMENTTYPE apartmentType;
         };
 
-        // Win32 method signature we're duplicating.
-        //HRESULT CreateDispatcherQueueController(
-        //  DispatcherQueueOptions options,
-        //  ABI::Windows::System::IDispatcherQueueController** dispatcherQueueController
-        //);
-        [DllImport("coremessaging.dll", EntryPoint = "CreateDispatcherQueueController", CharSet = CharSet.Unicode)]
+        [DllImport("coremessaging.dll")]
         internal static extern IntPtr CreateDispatcherQueueController(DispatcherQueueOptions options,
```

Comment:
>IntPtr [](start = 31, length = 6)

Return type is wrong. I'm guessing this would fail on 32 bit processes. Should be an HRESULT. Returning an int is ok, and you should check that it returns 0 and throw an exception if it doesn't. See Marshal.ThrowExceptionForHR for an easy way to come up with an exception.